### PR TITLE
Crowdloan Claim Pallet V2 -Alternative to  #385

### PR DESCRIPTION
**Pull request by mustermeiszer**
This PR is an alternative to #385. 
The difference here are that the pallets need to be initialized separately. Hence, the `Reward` trait has a smaller interface and only provides a rewarding functionality. 
Furthermore, the claim pallet is changed, in order to provide the possibility to initialize it multiple times.

### CODE REVIEW

File: pallets/crowdloan-claim/src/lib.rs [hunk 1]
```diff
@@ -320,14 +344,48 @@ pub mod pallet {
...
+        StorageMap<_, Blake2_128Concat, (T::RelayChainAccountId, Index), bool>;
+
+    #[pallet::type_value]
+    pub fn OnIndexEmpty() -> Index {
+        0
+    }
+
+    #[pallet::storage]
+    #[pallet::getter(fn curr_index)]
+    pub type CurrIndex<T: Config> = StorageValue<_, Index, ValueQuery, OnIndexEmpty>;
```
Comment (vedhavyas) [inline pallets/crowdloan-claim/src/lib.rs:356]: is it not possible to give the default here like pallet V1 ?

File: pallets/crowdloan-claim/src/mock.rs [hunk 2]
```diff
@@ -63,6 +63,26 @@ impl WeightInfo for MockWeightInfo {
     fn claim_reward() -> Weight {
         0 as Weight
     }
+
+    fn set_lease_start() -> u64 {
+        0 as Weight
```
Comment (vedhavyas) [inline pallets/crowdloan-claim/src/mock.rs:68]: so benchmarking is done in a different PR ?

**Review by vedhavyas — APPROVED** [review]
Some questions but LGTM otherwise.

Thanks!

File: pallets/crowdloan-claim/src/mock.rs (hunk 2 above)
Comment (mustermeiszer) [reply pallets/crowdloan-claim/src/mock.rs:68]: Yes. Once this is through I will add a PR for this pallet in parachain, where I integrate the benchmark. Wanted to close this first, finally...

File: pallets/crowdloan-claim/src/lib.rs (hunk 1 above)
Comment (mustermeiszer) [reply pallets/crowdloan-claim/src/lib.rs:356]: To be honest, I don't know. Thought, that this replaces the default, as you can also put arbitrary logic in the function like with the build closure.

Comment (vedhavyas) [reply pallets/crowdloan-claim/src/lib.rs:356]: hmm. okay. That one seems much simple.
